### ISSUE TITLE
Incorrect MSTUMP Test Results

### ISSUE THREAD

**Issue by seanlaw** [TDAmeritrade/stumpy#146]
After merging #142 (which passed tests on Azure), I downloaded the new code and tried running the tests on various machines. I noticed that `tests/test_mstump.py` was failing when the input was randomly generated. This code snipper should be able to reproduce the failed test:

```
def test_mstump(T, m):
        T = np.array(
            [
                [
                    943.05068649,
                    239.38617809,
                    598.8576716,
                    940.10542282,
                    -257.48207855,
                    -327.69698105,
                    -422.73486556,
                    -515.66419683,
                    -738.51165339,
                    251.35295679,
                ],
                [
                    -362.65061832,
                    994.99535864,
                    -95.76512357,
                    87.43404547,
                    -864.98178883,
                    350.00063995,
                    -87.65017707,
                    708.04802098,
                    420.08440992,
                    469.71268701,
                ],
                [
                    -267.23624406,
                    -198.89118733,
                    38.26609812,
                    -210.49805755,
                    -175.39250349,
                    -590.46590949,
                    314.35972467,
                    -666.53859852,
                    245.56262824,
                    628.19924259,
                ],
            ]
        )

        excl_zone = int(np.ceil(m / 4))

        left_P, left_I = utils.naive_mstump(T, m, excl_zone)
        right_P, right_I = mstump(T, m)
        npt.assert_almost_equal(left_P, right_P)
```

**Comment by seanlaw** [comment]
@mexxexx I'm not sure if this is related to any of the changes that were made. Any chance you could take a look?

**Comment by seanlaw** [comment]
We should probably add this input as a unit test once we understand what is going on.

**Comment by seanlaw** [comment]
Okay, this is strange, this same code snippet succeeds on my laptop but not on another machine

**Comment by mexxexx** [comment]
It also succeeds on my machine, with an absolute error of 10^-13

**Comment by seanlaw** [comment]
Here'e the `left_P` and `right_P` output that it prints out:

```
[[1.88994576 1.9257502  2.20579762]
 [1.91056192 3.00313959 3.43743266]
 [3.01171953 3.20262456 3.47250755]
 [2.16047338 3.03425798 3.4968227 ]
 [1.88994576 1.9257502  2.20579762]
 [3.01171953 3.20262456 3.45696204]]

[[1.88994576 1.9257502  2.20579762]
 [1.91056192 3.00313959 3.43743266]
 [3.01171953 3.20262456 3.47250755]
 [2.16047338 3.03425798 3.4968227 ]
 [1.88994576 3.00313959 3.43743266]
 [3.01171953 3.20262456 3.45696204]]
```

**Comment by mexxexx** [comment]
When I run this test on my machine, the result is the following:

```
right_P
array([[0.84885263, 1.01241696, 1.30108876],
       [0.00766194, 0.64216018, 1.16324649],
       [0.9749291 , 1.20445752, 1.60955794],
       [0.44748054, 0.62521088, 0.91963731],
       [0.00329304, 0.12858288, 0.44570532],
       [0.04920799, 0.62521088, 0.91963731],
       [0.00329304, 0.12858288, 0.44570532],
       [0.00766194, 0.64216018, 1.16324649]])
```

This looks completely different. Could you print `right_P` on a machine where the tests pass?

**Comment by seanlaw** [comment]
> And on the other one, maybe deactivate numba and see if it still fails.

Good call on this one! I noticed that the "bad" machine was running `Numba` version `0.46.0` and my laptop was running `0.48.0`. After I updated the version to `0.48.0` on the "bad" machine then the tests passed!  I'm not sure what the difference is and why it would be causing issues but maybe we should update the `setup.py` to change the minimum `Numba` version to `numba >= 0.48.0`

**Comment by seanlaw** [comment]
Thank goodness for unit tests!

**Comment by seanlaw** [comment]
I ended up updating everything: `numpy`, `scipy`, `numba`, `pytest`, `pytest-cov`

**Comment by mexxexx** [comment]
That is still very wiered and it would be interesting to know what caused the issue

**Comment by seanlaw** [comment]
Agreed. I also noticed that installing via `conda install numba` doesn't work for running tests but `conda install -c conda-forge numba` (exact same version) seems stable (I've updated the docs to reflect this). `pip install numba` seems fine. I'll close this for now but the moral of the story is that one should probably update to the newest versions of `numpy`, `scipy`, and `numba` to be safe. Unfortunately, this isn't something that we can control.
